### PR TITLE
Disable tablet migration on repair

### LINKED ISSUE #3773: Stop tablet load balancing during repair
Repairing should happen only when tablet load balancing is disabled. Without that, it would be possible for some tablets (mapped to ranges) to escape from being repaired by migrating from not repaired range to an already repaired one.

But what should happen when repair is paused? If SM re-enables tablet load balancing, then it would need to repair interrupted table from scratch (note that SM repairs table by table). If SM leaves tablet load balancing as disabled, then it might cause problems when paused repair task is never resumed.

I think that re-enabling it and starting repair of interrupted table from scratch is safer, but it would be nice to have other opinions on that.

Ref https://github.com/scylladb/scylladb/issues/17435

cc: @karol-kokoszka @tzach @bhalevy

### PR DESCRIPTION
**Pull request by Michal-Leszczynski**
Fixes #3773

### CODE REVIEW

**Comment by karol-kokoszka** [comment]
@Michal-Leszczynski I know I overlooked in some previous review, but this part (not introduced in this PR directly)
```
// getTabletKs returns set of tablet replicated keyspaces.
func getTabletKs(ctx context.Context, client *Client) *strset.Set {
	out := strset.New()
	// Assume that errors indicate that endpoints rejected 'replication' param,
	// which means that given Scylla version does not support tablet API.
	// Other errors will be handled on other API calls.
	tablets, err := client.ReplicationKeyspaces(ctx, ReplicationTablet)
	if err != nil {
		return out
	}
	vnodes, err := client.ReplicationKeyspaces(ctx, ReplicationVnode)
	if err != nil {
		return out
	}
	// Even when both API calls succeeded, we need to validate
	// that the 'replication' param wasn't silently ignored.
	out.Add(tablets...)
	if out.HasAny(vnodes...) {
		return strset.New()
	}
	return out
}
```
Is problematic...

Errors are hidden.
Can you log these errors ?

File: pkg/service/repair/progress.go [hunk 1]
```diff
@@ -261,12 +262,12 @@ func (pm *dbProgressManager) emptyState() bool {
 	return true
 }
 
-func (pm *dbProgressManager) GetCompletedRanges(keyspace, table string) []scyllaclient.TokenRange {
+func (pm *dbProgressManager) GetCompletedRanges(keyspace, table string) (doneRanges []scyllaclient.TokenRange, allRangesCnt int) {
```
Comment (karol-kokoszka) [inline pkg/service/repair/progress.go:265]: What you need named return parameters for ?
They are not used.

**Review by karol-kokoszka — COMMENTED** [review]
Please address the general comment about logging error and it's good to go.

File: pkg/service/repair/progress.go (hunk 1 above)
Comment (Michal-Leszczynski) [reply pkg/service/repair/progress.go:265]: It was a tip from linter. It suggests to name returned values when their type doesn't make it obvious.

**Comment by Michal-Leszczynski** [comment]
@karol-kokoszka the logging comment was addressed. Could you take a look?

**Comment by Michal-Leszczynski** [comment]
@karol-kokoszka could you take a look?

File: pkg/scyllaclient/ring_describer.go [hunk 2]
```diff
@@ -18,6 +18,9 @@ type RingDescriber interface {
 	Reset(ctx context.Context)
 	// IsTabletKeyspace returns true if keyspace has tablet replication.
 	IsTabletKeyspace(keyspace string) bool
+	// ControlTabletLoadBalancing runs scyllaclient ControlTabletLoadBalancing
+	// only when there are any tablet tables detected.
```
Comment (karol-kokoszka) [inline pkg/scyllaclient/ring_describer.go:22]: Please change the comment. To inform that is about enabling/disabling tablets load balancing.

**Review by karol-kokoszka — APPROVED** [review]
LGTM !

**Pull request merged**